app/routes/administrative-units/administrative-unit/sites/site.js

```diff
@@ -0,0 +1,25 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+export default class AdministrativeUnitsAdministrativeUnitSitesSiteRoute extends Route {
+  @service store;
+
+  async model() {
+    let administrativeUnit = this.modelFor(
+      'administrative-units.administrative-unit'
+    );
+
+    let { siteId: siteId } = this.paramsFor(
```

Comment:
paramsFor shouldn't be needed anymore, you can now directly access the params inside the model hook. The first argument to the is an object with all the params.

->
`async model({ siteId }) { //... }`

Reply:
Can I have more details about the reason why is not anymore necessary?

Reply:
Sure. The router.js file shows that the `:siteId` dynamic segment was added to the `administrative-units.administrative-unit.sites.site` route. This means that that route will receive the value for that param as the first argument to the model hook (which is what I suggested here).

Before, `paramsFor` was needed since the fetching was done from the `administrative-units.administrative-unit.sites.site.index` route. That's not the route that defines the param so we don't receive it directly in the model hook.

Reply:
Clear! Thank you!